### DISCUSSION TITLE
Update alert only if build is newer

### DISCUSSION THREAD

**Discussion by oroboros** [valinet/ExplorerPatcher#710] [Q&A] [answered]
If a pre-release user chooses to not receive pre-release updates (from settings), checking for update shouldn't alert them of an update unless it's a newer build. 

I'm on 22000.434.41.11

EDIT: Just to be clear: I updated to the pre-release just now (22000.434.41.11) and it is alerting me of an update, pointing me to (22000.376.40.21).

**Comment by valinet** [comment, marked as answer]
This has been discussed before. Version numbers are meaningless for the updater, they’re only there for humans to kind of keep track. As you have seen, there may be different builds with the same version. If the updater sees a different hash for the remote file compared to the local file, it deems it an update and that’s it.

Also, pre-releases are for enthusiasts. The hope is that people use these, identify problems and help solve them in the upstream, working towards a stable, bug free pre-release that becomes a release. For such an use case, there is no need to go back to a previous release.

**Reply by oroboros** [reply]
I can totally see how a non-expert user might want to use a new feature in one of the pre-releases right away but doesn't want to continue staying on the alpha/beta channel forever (as was my case). Just giving you this user's feedback! Thanks again!

**Reply by valinet** [reply]
I understand this. You can also disable the updates. When a new “stable” release is made, you switched them back on and update to that. Stable releases anyway come at the end of the month mostly…

**Reply by valinet** [reply]
You can also check these out, where this has been discussed previously:

* https://github.com/valinet/ExplorerPatcher/wiki/Configure-updates
* https://github.com/valinet/ExplorerPatcher/discussions/540

In general, it’s good to reuse existing threads on similar topics rather than to open new ones. Thanks.

**Reply by valinet** [reply]
I have implemented functionality that treats this use case and does not suggest the older stable version as an update when the user is using a pre-release version but has switched  back to the stable update channel in the latest pre-release (22000.434.41.12). Indeed, it's better this way. Additionally, a new feature is also that the notification that gets displayed when an update is available now includes the version number of the update.

Thanks for the report.
